src/app/content/highlights/hooks/locationChange.ts

```diff
@@ -16,16 +17,28 @@ const hookBody = (services: MiddlewareAPI & AppServices) => async() => {
     return;
   }
 
-  const highlights = await highlightClient.getHighlights({
-    perPage: 100,
-    scopeId: book.id,
-    sourceIds: [page.id],
-    sourceType: GetHighlightsSourceTypeEnum.OpenstaxPage,
-  });
+  const [highlights, totalCounts] = await Promise.all([
+    await highlightClient.getHighlights({
+      perPage: 100,
+      scopeId: book.id,
+      sourceIds: [page.id],
+      sourceType: GetHighlightsSourceTypeEnum.OpenstaxPage,
+    }),
+    await highlightClient.getHighlightsSummary({
```

Comment:
we don't have to do this on every navigation, you could move it to an initializer that only happens once on app init. or you could put a check here that only queries it if its not already there

Reply:
I've added check for equality between those two objects which can be helpful in cases when user is working on multiple tabs

Reply:
there are way more problems with supporting highlighting in multiple tabs, i would rather not have the extra network request on every navigation

Reply:
Ok, now I'm doing:
```
if (Object.keys(totalCountsInState).length > 0) { return; }
```
however there is something strange going on, because `totalCountsInState` is always an empty object even if state is propagated with data. When I console.log state in `locationChange` then I can see that `totalCountsPerLocation` are propagated, but `..perPage` are not.
Any ideas?